website/integrations: gitguardian: add (#23587)
goauthentik/authentik: website/integrations/security/gitguardian/index.mdx

sidebar_label: GitGuardian
support_level: community
---

import SAMLProvider20265Warning from "../../_saml-provider-2026-5-warning.mdx";

## What is GitGuardian?

> GitGuardian is a security platform that helps teams detect and remediate exposed secrets, monitor software development lifecycle credentials, and govern non-human identities.
>
> -- https://www.gitguardian.com/

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.
- `gitguardian.company` is the FQDN of the GitGuardian installation, when using self-hosted GitGuardian.

:::info
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

:::info GitGuardian requirements
GitGuardian SAML SSO can be configured by the Owner or any Manager of the workspace. GitGuardian supports Just-In-Time (JIT) provisioning for users who are authorized to access the authentik application. SCIM provisioning requires SAML SSO to be configured first.
:::

## authentik configuration

> To support the integration of GitGuardian with authentik, you need to create property mappings and an application/provider pair in authentik. If you also want to provision GitGuardian users and teams from authentik, create the optional SCIM property mappings.

Reviewer: ```suggestion
To support the integration of GitGuardian with authentik, you need to create property mappings and an application/provider pair in authentik. Optionally, to provision GitGuardian users and teams from authentik, also create the SCIM property mappings and a SCIM provider.
```

        - Temporarily set the **ACS URL** and **Audience** to `https://temp.temp`.
        - Under **Advanced protocol settings**:
            - Select any available **Signing Certificate**.
            - Set **NameID Property Mapping** to `authentik default SAML Mapping: Email`.
            - Set **Default NameID Policy** to `Email address`.
            - Add the `GitGuardian first name` and `GitGuardian last name` property mappings that you created in the previous section.
    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/bindings-overview/) (policy, group, or user) to manage the listing and access to applications on a user's **Application Dashboard** page.

3. Click **Submit** to save the new application and provider.

### Download the signing certificate

1. Navigate to **Applications** > **Providers** and click the name of the SAML provider that you created.
2. Under **Related objects** > **Download signing certificate**, click **Download**. This certificate file is required in the GitGuardian SSO setup flow.

> ### Create SCIM property mappings _(optional)_

Reviewer: Maybe it would be better to split the SCIM setup apart by using tabs?

> GitGuardian SCIM provisioning expects email-based user identifiers. Make sure each authentik user that should be provisioned to GitGuardian has an email address.

Reviewer: ```suggestion
GitGuardian SCIM provisioning uses email addresses as user identifiers, so only users with emails can be provisioned. Ensure every authentik user you plan to provision to GitGuardian has a valid email address configured.
```
